oauth, message controller 작성

KTH-96 opened the pull request.

KTH-96 committed:
feat oauth 로그인 Controller 생성

- moveOauthLoginPage 로그인 페이지로 이동
- oauthLogin 토큰 인가 코드로 토큰 발급받아 넘겨준다

KTH-96 committed:
feat 사용자가 알림을 원하는 정보 받기

- jackson 상속으로 type값에 따라 다른 Dto 생성

f-lab-lyan commented on api/src/main/java/com/kth/mssage/info/web/dto/request/RequestInfoDto.java:
```diff
@@ -0,0 +1,26 @@
+        @JsonSubTypes.Type(name = "weather", value = RequestWeatherDto.class)
+})
+public class RequestInfoDto {
+
```
```suggestion
    private String type;
```

f-lab-lyan commented on api/src/main/java/com/kth/mssage/info/web/controller/InfoController.java:
```diff
@@ -0,0 +1,18 @@
+@RestController
+public class InfoController {
+
+    @GetMapping("/message")
```
```suggestion
    @PostMapping("/message")
```

f-lab-lyan commented:
1. Controller에 IntegerationTest를 추가해주세요. Json이 제대로 파싱되는지 확인할 수 있으면 좋을 것 같습니다.
2. Spring의 ThreadPool모델에 대해서 얘기해보죠. 싱글톤을 포함해서

KTH-96 committed:
build lombok 추가

KTH-96 committed:
refactor api 파싱 구조 변경 및 변수에 final 추가

- 챗봇 api맞는 구조로 변경

KTH-96 committed:
refactor Json 역직렬화시에 생성자 또는 팩토리 메서드 찾지 못하는 현상 제거

- @JsonCreator로 데이터 파싱

KTH-96 committed:
feat 챗봇 스킬 반환값 만들기

KTH-96 committed:
docs out 폴더 지우기

KTH-96 committed:
test mock mvc를 이용해 Json 데이터 파싱 테스트 작성

KTH-96 committed:
refactor 컨트롤러 이름 및 패키지 구조변경

- 기존 InfoController를 SimpleController로 변경
- api 응답값 수정(SimpleTextContentDto클래스 추가)
- test json데이터 수정

KTH-96 committed:
feat text card controller생성

- 응답 예제 추가

KTH-96 committed:
test text card 테스트 추가

f-lab-lyan commented on api/build.gradle:
```diff
@@ -12,4 +12,10 @@ dependencies {
     implementation project(':domain')
     implementation project(':infra')
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.8.1'
```
```suggestion
    testImplementation 'org.junit.jupiter:junit-jupiter'
```

f-lab-lyan commented on api/build.gradle:
```diff
@@ -12,4 +12,10 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.8.1'
+
+    compileOnly 'org.projectlombok:lombok:1.18.30'
```
```suggestion
    compileOnly 'org.projectlombok:lombok'
```

f-lab-lyan commented on api/src/main/java/com/kth/mssage/info/web/controller/SimpleTextController.java:
```diff
@@ -0,0 +1,36 @@
+    @ResponseStatus(value = HttpStatus.OK)
+    @PostMapping("/message")
+    public ResponseResultDto requestMessageInfo(@RequestBody RequestActionDto requestActionDto) {
+        WeatherDto weatherDto = (WeatherDto) requestActionDto.getAction().getParamDto();
```
```suggestion
        ParamDto paramDto = requestActionDto.getAction().getParamDto();
        if (paramDto instanceOf WeatherDto) {
            WeatherDto weatherDto = (WeatherDto) ....
        }
```

f-lab-lyan commented on api/src/main/java/com/kth/mssage/info/web/dto/request/ActionDto.java:
```diff
@@ -0,0 +1,18 @@
+
+@Getter
+@Builder
+public class ActionDto {
+
+    private final ParamDto paramDto;
+
```
```suggestion
public class ActionDto<T extends ParamDto> {

    private final T dto;

```

f-lab-lyan commented on api/src/main/java/com/kth/mssage/info/web/dto/request/RequestActionDto.java:
```diff
@@ -0,0 +1,18 @@
+
+@Getter
+@Builder
+public class RequestActionDto {
+
+    private final ActionDto action;
+
```
```suggestion
public class RequestActionDto<T extends ParamDto> {

    private final ActionDto<T> action;

```

KTH-96 committed:
refactor 코드 리펙토링

- method readJsonFile resources 하위 파일 읽는 메소드 변경
- 상수 인터페이스 삭제
